opengemini-client-common/src/main/java/io/opengemini/client/common/BaseClient.java

```diff
@@ -56,21 +56,22 @@ public BaseClient(BaseConfiguration conf) {
      * Start schedule task(period 10s) to ping all server url
      */
     private void startHealthCheck(ScheduledExecutorService healthCheckSchedule) {
-        healthCheckSchedule.scheduleWithFixedDelay(new Runnable() {
-            @Override
-            public void run() {
-                for (Endpoint url : serverUrls) {
+        healthCheckSchedule.scheduleWithFixedDelay(() -> {
+            for (Endpoint url : serverUrls) {
+                try {
+                    URL urls = new URL(url.getUrl() + getPingUrl());
+                    HttpURLConnection connection = (HttpURLConnection) urls.openConnection();
                     try {
-                        URL urls = new URL(url.getUrl() + getPingUrl());
-                        HttpURLConnection connection = (HttpURLConnection) urls.openConnection();
                         connection.setRequestMethod("GET");
                         connection.setConnectTimeout(2000);
                         connection.connect();
                         int responseCode = connection.getResponseCode();
```

Comment:
connection's inputstream should be closed too.

Reply:
the inputsream had been closed insite HttpURLConnection.disconnect()
```
    public void disconnect() {
        ......
        if (http != null) {
            if (inputStream != null) {
                ......
                try {
                    inputStream.close();
                } catch (IOException ioe) { }
                // if the connection is persistent it may have been closed
                // or returned to the keep-alive cache. If it's been returned
                // to the keep-alive cache then we would like to close it
                // but it may have been allocated
                if (ka) {
                    hc.closeIdleConnection();
                }
            }
```